File: package/main/src/URL/isAbsoluteUrl.ts
/**
 * Checks whether a URL string is absolute (RFC 3986).
 *
 * An absolute URL starts with a scheme followed by a colon,
 * where the scheme begins with a letter and may contain
 * letters, digits, plus, hyphen, or period.
 *
 * @param url - The URL string to check
 * @returns True if the URL is absolute, false otherwise
 *
 * @example
 * ```typescript
 * isAbsoluteUrl("https://example.com");  // true
 * isAbsoluteUrl("ftp://files.example");  // true
 * isAbsoluteUrl("/path/to/page");        // false
 * isAbsoluteUrl("relative/path");        // false
 * isAbsoluteUrl("mailto:user@host");     // true
 * ```
 */
export const isAbsoluteUrl = (url: string): boolean => {
  return /^[a-z][a-z\d+\-.]*:/i.test(url);
};

File: package/umt_python/src/url/is_absolute_url.py
import re


def is_absolute_url(url: str) -> bool:
    """Checks whether a URL string is absolute (RFC 3986).

    An absolute URL starts with a scheme followed by a colon,
    where the scheme begins with a letter and may contain
    letters, digits, plus, hyphen, or period.

    :param url: The URL string to check
    :return: True if the URL is absolute, False otherwise
    """
    return bool(re.match(r"^[a-z][a-z\d+\-.]*:", url, re.IGNORECASE))
